packages/mc-html-template/lib/load-env.spec.js

```diff
@@ -0,0 +1,107 @@
+const fs = require('fs');
+const loadEnv = require('./load-env');
+
+jest.mock('fs');
+
+const minimalEnvConfig = {
+  applicationName: 'testa-application',
+  frontendHost: 'test-frontend-host',
+  mcApiUrl: 'test-mc-api-url',
+  location: 'test-location',
+  env: 'test-env',
+  cdnUrl: 'test-cdn-url',
+};
+
+const fsMocks = {
+  withMissingKey: JSON.stringify({
+    ...minimalEnvConfig,
+    cdnUrl: undefined,
+  }),
+  withoutMissingKey: JSON.stringify(minimalEnvConfig),
+  withPlaceholders: JSON.stringify({
+    ...minimalEnvConfig,
+    apiUrl: '${env:API_URL}',
+  }),
+  withMultiplePlaceholders: JSON.stringify({
+    ...minimalEnvConfig,
+    apiUrl: 'http://${env:API_HOSTNAME}.${env:API_TLD}',
+  }),
+};
+
+describe('with missing required environment config values', () => {
+  it('should throw an error and not return the parsed config', () => {
+    fs.readFileSync.mockReturnValue(fsMocks.withMissingKey);
+
+    expect(() => loadEnv(undefined, { disableCache: true })).toThrow(
+      /Missing 'cdnUrl' required configuration field/
+    );
+  });
+});
+
+describe('with all required environment config values', () => {
+  it('should return the parsed config and all values', () => {
+    fs.readFileSync.mockReturnValue(fsMocks.withoutMissingKey);
+
+    expect(loadEnv(undefined, { disableCache: true })).toEqual(
+      expect.objectContaining(minimalEnvConfig)
+    );
+  });
+  it('should return default values', () => {
+    fs.readFileSync.mockReturnValue(fsMocks.withoutMissingKey);
+
+    expect(loadEnv(undefined, { disableCache: true })).toEqual(
+      expect.objectContaining({ servedByProxy: false })
+    );
+  });
+});
+
+describe('with environment variable placeholders', () => {
+  describe('when all placeholders are defined on environment', () => {
+    it('should return the parsed config and all values without placeholders', () => {
+      process.env.API_URL = 'test-api-url';
+
+      fs.readFileSync.mockReturnValue(fsMocks.withPlaceholders);
+
+      expect(loadEnv(undefined, { disableCache: true })).toEqual(
+        expect.objectContaining(minimalEnvConfig)
+      );
+    });
+    it('should return the parsed config and all values with placeholders', () => {
+      process.env.API_URL = 'test-api-url';
+
+      fs.readFileSync.mockReturnValue(fsMocks.withPlaceholders);
```

Comment:
Isn't this the same as the test above? 🤔

Reply:
Not really. I wanted to split testing that both 1. the variables without placeholders and 2. the onces with placeholders are defined. Helped me when building it to spot errors.

Reply:
Sure but those 2 tests are basically the same. I would have expected to just enhance the assertion, like:

```js
expect(loadEnv(undefined, { disableCache: true })).toEqual(
  expect.objectContaining({
    ...minimalEnvConfig,
    apiUrl: process.env.API_URL,
  })
);
```

Maybe it's just me that finds this a bit confusing 🙃

Reply:
This is what I meant: https://github.com/commercetools/merchant-center-application-kit/pull/1380/commits/ccf98f7d76003e13a4a9c8040075c7ea5a3057b5

If you had something else in mind please forgive me and feel free to revert.

Reply:
Yeah I understood before I had that at first too. I felt it's two cases to test: have the ones with placeholders and those without. It was nice to see one spec failing when working on placeholders. Anway. I don't mind :)